sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -1254,6 +1255,1196 @@ raw_send_store(simd<T1, n1> msgSrc0, uint32_t exDesc, uint32_t msgDesc,
+  constexpr lsc_data_size _DS = detail::finalize_data_size<T, DS>();
+  constexpr detail::lsc_vector_size _VS = detail::to_lsc_vector_size<NElts>();
+  constexpr detail::lsc_data_order _Transposed =
+      detail::lsc_data_order::nontranspose;
+  constexpr detail::lsc_atomic_op _Op = detail::to_lsc_atomic_op<Op>();
+  return __esimd_lsc_xatomic_slm_1<T, _Op, L1H, L3H, _AddressScale, _ImmOffset,
+                                   _DS, _VS, _Transposed, N>(
+      pred.data(), offsets.data(), src0.data());
+}
+
+/// SLM atomic.
+/// Supported platforms: XeHP_SDV, DG2, PVC
+/// VISA instruction: lsc_atomic_<OP>.slm
+///
+/// @tparam T is element type.
+/// @tparam Op is operation type.
+/// @tparam NElts is the number of elements per address.
+/// @tparam DS is the data size.
+/// @tparam L1H is L1 cache hint.
+/// @tparam L3H is L3 cache hint.
+/// @tparam N is the number of channels (platform dependent).
+/// @param offsets is the zero-based offsets.
+/// @param src0 is the first atomic operand.
+/// @param src1 is the second atomic operand.
+/// @param pred is predicates.
+///
+template <typename T, atomic_op Op, uint8_t NElts = 1,
+          lsc_data_size DS = lsc_data_size::default_size,
+          cache_hint L1H = cache_hint::none, cache_hint L3H = cache_hint::none,
+          int N>
+__ESIMD_API simd<T, N * NElts>
+lsc_slm_atomic_update(simd<uint32_t, N> offsets, simd<T, N * NElts> src0,
+                      simd<T, N * NElts> src1, simd_mask<N> pred) {
+  detail::check_lsc_vector_size<NElts>();
+  detail::check_lsc_data_size<T, DS>();
+  detail::check_lsc_atomic<Op, 2>();
+  constexpr uint16_t _AddressScale = 1;
+  constexpr int _ImmOffset = 0;
+  constexpr lsc_data_size _DS = detail::finalize_data_size<T, DS>();
+  constexpr detail::lsc_vector_size _VS = detail::to_lsc_vector_size<NElts>();
+  constexpr detail::lsc_data_order _Transposed =
+      detail::lsc_data_order::nontranspose;
+  constexpr detail::lsc_atomic_op _Op = detail::to_lsc_atomic_op<Op>();
+  return __esimd_lsc_xatomic_slm_2<T, _Op, L1H, L3H, _AddressScale, _ImmOffset,
+                                   _DS, _VS, _Transposed, N>(
+      pred.data(), offsets.data(), src0.data(), src1.data());
+}
+
+/// Accessor-based atomic.
+/// Supported platforms: XeHP_SDV, DG2, PVC
+/// VISA instruction: lsc_atomic_<OP>.ugm
+///
+/// @tparam T is element type.
+/// @tparam Op is operation type.
+/// @tparam NElts is the number of elements per address.
+/// @tparam DS is the data size.
+/// @tparam L1H is L1 cache hint.
+/// @tparam L3H is L3 cache hint.
+/// @tparam N is the number of channels (platform dependent).
+/// @tparam AccessorTy is the \ref sycl::accessor type.
+/// @param acc is the SYCL accessor.
+/// @param offsets is the zero-based offsets.
+/// @param pred is predicates.
+///
+template <typename T, atomic_op Op, uint8_t NElts = 1,
+          lsc_data_size DS = lsc_data_size::default_size,
+          cache_hint L1H = cache_hint::none, cache_hint L3H = cache_hint::none,
```

Comment:
L1 cache hint shall be either default or uncached

Reply:
> Does it apply to all atomic operations (excluding SLM)?

Yes

> Does it apply to all atomic operations (excluding SLM)? Should `L1H` parameter be removed atomic templates?

I don't really know. Probably we need just a static assert

Reply:
Removed L1 cache hint from all atomics.